Sources/Customization/DefaultDatafileHandler.swift

```diff
@@ -21,11 +21,11 @@ open class DefaultDatafileHandler: OPTDatafileHandler {
     public var endPointStringFormat = "https://cdn.optimizely.com/datafiles/%@.json"
     
     // thread-safe lazy logger load (after HandlerRegisterService ready)
-    private var loggerInstance: OPTLogger?
+    private let threadSafeLogger = ThreadSafeLogger()
     var logger: OPTLogger {
-        return OPTLoggerFactory.getLoggerThreadSafe(&loggerInstance)
+        return threadSafeLogger.logger
     }
```

Comment:
Although the code looks more readable this way, but how is this more thread safe? just curious! Instead of passing instance to a function, we have it inside a separate class now but the overall functionality remains the same.

Reply:
"&loggerInstance" in old line 26 is read in outside of thread-safe sync (and passed to sync), so potentially can be a source of a race condition.